cg/store/filters/status_order_filters.py

```diff
@@ -6,27 +6,31 @@
 from cg.store.models import Order
 
 
-def filter_orders_by_workflow(orders: Query, workflow: str, **kwargs) -> Query:
-    """Return orders filtered on workflow."""
+def filter_orders_by_workflow(orders: Query, workflow: str | None, **kwargs) -> Query:
     return orders.filter(Order.workflow == workflow) if workflow else orders
 
 
-def filter_orders_by_id(orders: Query, id: int, **kwargs) -> Query:
-    """Return orders filtered on id."""
-    return orders.filter(Order.id == id)
+def filter_orders_by_id(orders: Query, id: int | None, **kwargs) -> Query:
+    return orders.filter(Order.id == id) if id else orders
 
 
-def apply_order_filters(
-    filter_functions: list[Callable], orders: Query, id: int = None, workflow: str = None
-) -> Query:
-    """Apply filtering functions to the order queries and return filtered results."""
-    for filter_function in filter_functions:
-        orders: Query = filter_function(orders=orders, id=id, workflow=workflow)
-    return orders
+def filter_by_limit(orders: Query, limit: int | None, **kwargs) -> Query:
+    return orders.limit(limit) if limit else orders
```

Comment:
Something feels a bit off with calling "limit" a filter, given that it's more a truncation.

Reply:
Renamed